src/lib.rs

```diff
@@ -39,6 +48,7 @@ pub mod blake2b;
 pub mod blake2s;
 
 /// Runs the self-test for both BLAKE2b and BLAKE2s.
+#[cfg(test)]
```

Comment:
Hiding the selftest functions would change the crate API in the `cfg(std)` case.

Reply:
Ah silly me, I didn't pay attention to the fact that they were `pub`. Though if we want to keep them in the API, having them return `ArrayVec` would be a breaking change also. What's the right move?

Reply:
Maybe changing the API in the `no_std` case is the right thing to do? All current callers should default to enabling `std`, and so there shouldn't be any breaking changes?

Reply:
The three selftest functions already return nothing, see for instance https://docs.rs/blake2-rfc/0.2.17/blake2_rfc/fn.selftest.html. No need to change the API.

The selftest_seq function is internal, it might be `pub` but is within a non-`pub` module, so it's API is not a concern.